internal/ptx/obfs4.go

```diff
@@ -0,0 +1,168 @@
+package ptx
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"path/filepath"
+	"time"
+
+	pt "git.torproject.org/pluggable-transports/goptlib.git"
+	"github.com/ooni/probe-cli/v3/internal/runtimex"
+	"gitlab.com/yawning/obfs4.git/transports/base"
+	"gitlab.com/yawning/obfs4.git/transports/obfs4"
+)
+
+// DefaultTestingOBFS4Bridge is a factory that returns you
+// an OBFS4Dialer configured for the bridge we use by default
+// when testing. Of course, given the nature of obfs4, it's
+// not wise to use this bridge in general. But, feel free to
+// use this bridge for integration testing of this code.
+func DefaultTestingOBFS4Bridge() *OBFS4Dialer {
+	// TODO(bassosimone): this is a public working bridge we have found
+	// with @hellais. We should ask @cohosh whether there's some obfs4 bridge
+	// dedicated to integration testing that we should use instead.
```

Comment:
Cc: @cohosh

Reply:
We don't run any bridges dedicated to testing. I'd recommend just using one of our default bridges. Here's the list:

https://gitlab.torproject.org/tpo/applications/tor-browser-build/-/blob/master/projects/tor-browser/Bundle-Data/PTConfigs/bridge_prefs.js

Reply:
Maybe instead of a bridge that comes from bridgeDB we should use instead one of the default bridges used by tor-browser.

Here is the list of all the bridges: https://github.com/ooni/sysadmin/blob/master/ansible/roles/probe-services/templates/tor_targets.json#L122

Reply:
@cohosh @hellais thanks!

The bridge _used to be_ part of the Tor Browser bridges from what I recall. Maybe now it has been removed from that pool? (In any case, I'd like to reassure you all that this is not a bridge we got from Bridge DB.)

I'll take one from the `bridge_prefs.js`!

Reply:
Yes @bassosimone is right. That bridge used to be part of Tor Browser, but was dropped in the following commit: https://gitlab.torproject.org/tpo/applications/tor-browser-build/-/commit/e26e91bef8bd8d04d79bdd69f087efd808bc925d.

My bad for hinting at the fact it might have come from bridgeDB.

Reply:
🙏 !